ffi/derive/tests/ui_fail/self_consuming_method.rs

```diff
@@ -0,0 +1,31 @@
+use iroha_ffi::{ffi_bindgen, FfiResult};
+use std::mem::MaybeUninit;
+
+struct FfiStructBuilder;
+struct FfiStruct;
+
+#[ffi_bindgen]
+impl FfiStructBuilder {
+    pub fn new() -> Self {
+        FfiStruct
+    }
+
```

Comment:
This is a bit clearer, but I'd add a doc-comment nonetheless.

Reply:
added a comment. Again, this is just a test guarding against a feature we have no use of